### PR TITLE
Update environments documentation links.

### PR DESCRIPTION
**Pull request by igorborgest**
The environment documentation links are outdated in the CONTRIBUTION guide. So I've updated that.

### CODE REVIEW

File: CONTRIBUTING.rst [hunk 1]
```diff
@@ -503,10 +503,10 @@ Development Environments
 There are two environments, available on Linux and macOS, that you can use to
 develop Apache Airflow:
 
--   `Local virtualenv development environment <#local-virtualenv-development-environment>`_
+-   `Local virtualenv development environment <https://github.com/apache/airflow/blob/main/LOCAL_VIRTUALENV.rst>`_
```
Comment (jedcunningham) [inline CONTRIBUTING.rst]:
```suggestion
-   `Local virtualenv development environment <LOCAL_VIRTUALENV.rst>`_
```

These can be relative.

File: CONTRIBUTING.rst [hunk 2]
```diff
@@ -503,10 +503,10 @@ Development Environments
...
     that supports running unit tests and can be used in your IDE.
 
--   `Breeze Docker-based development environment <#breeze-development-environment>`_ that provides
+-   `Breeze Docker-based development environment <https://github.com/apache/airflow/blob/main/BREEZE.rst>`_ that provides
```
Comment (jedcunningham) [inline CONTRIBUTING.rst]:
```suggestion
-   `Breeze Docker-based development environment <BREEZE.rst>`_ that provides
```

File: CONTRIBUTING.rst [hunk 3]
```diff
@@ -564,7 +564,7 @@ Limitations:
     real unit tests. Technically, to run integration tests, you can configure
     and install the dependencies on your own, but it is usually complex.
     Instead, you are recommended to use
-    `Breeze development environment <#breeze-development-environment>`__ with all required packages
+    `Breeze development environment <https://github.com/apache/airflow/blob/main/BREEZE.rst>`__ with all required packages
```
Comment (jedcunningham) [inline CONTRIBUTING.rst]:
```suggestion
    `Breeze development environment <BREEZE.rst>`__ with all required packages
```

**Review by jedcunningham — APPROVED** [review]

**Comment by jedcunningham** [comment]
Thanks again @igorborgest 🍺

**Comment by potiuk** [comment]
A LOT !

**Pull request merged**